### PR TITLE
service implementation

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/mate/jdbc/Factory/FactorySingleton.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package mate.jdbc.Factory;
+
+import mate.jdbc.service.CarService;
+import mate.jdbc.service.impl.CarServiceImpl;
+
+public class FactorySingleton {
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/Factory/FactorySingleton.java]: @EliosPride rename -> CarServiceFactory

File: src/main/java/mate/jdbc/Factory/FactorySingleton.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package mate.jdbc.Factory;
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/Factory/FactorySingleton.java]: @EliosPride package name starts from lowercase by naming convension

File: src/main/java/mate/jdbc/controller/CarAddController.java [hunk 3]
```diff
@@ -21,10 +23,11 @@
 public class CarAddController extends HttpServlet {
     private final CarDao carDao = (CarDao) InjectorUtils.getInstance(CarDao.class);
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/controller/CarAddController.java]: @EliosPride you don't need this anymore here

File: src/main/java/mate/jdbc/controller/DriverAccountController.java [hunk 4]
```diff
@@ -16,11 +18,12 @@
 @WebServlet("/driver-account")
 public class DriverAccountController extends HttpServlet {
     private final CarDao carDao = (CarDao) InjectorUtils.getInstance(CarDao.class);
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/controller/DriverAccountController.java]: @EliosPride remove

File: src/main/java/mate/jdbc/service/impl/CarServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.dao.impl.CarDaoImpl;
+import mate.jdbc.model.Car;
+import mate.jdbc.service.CarService;
+
+import java.util.List;
+import java.util.Optional;
+
+public class CarServiceImpl implements CarService {
+    CarDao carDao = new CarDaoImpl();
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/service/impl/CarServiceImpl.java]: @EliosPride 
1. private final
3. use injector

File: src/main/java/mate/jdbc/controller/CarAddController.java (hunk 3 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/CarAddController.java]: @al-pavlenko 2

File: src/main/java/mate/jdbc/Factory/FactorySingleton.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/Factory/FactorySingleton.java]: @al-pavlenko 2

File: src/main/java/mate/jdbc/controller/DriverAccountController.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/controller/DriverAccountController.java]: @al-pavlenko 2

File: src/main/java/mate/jdbc/service/impl/CarServiceImpl.java (hunk 5 above)
Comment (EliosPride) [reply src/main/java/mate/jdbc/service/impl/CarServiceImpl.java]: @al-pavlenko 2

File: src/main/java/mate/jdbc/factory/FactorySingleton.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package mate.jdbc.factory;
+
+import mate.jdbc.service.CarService;
+import mate.jdbc.service.impl.CarServiceImpl;
+
+public class FactorySingleton {
+    private static CarService carService;
+
+    public static CarService getCarService() {
```
Comment (al-pavlenko) [inline src/main/java/mate/jdbc/factory/FactorySingleton.java]: also rename to -> getInstance()

File: src/main/java/mate/jdbc/Factory/FactorySingleton.java (hunk 1 above)
Comment (al-pavlenko) [reply src/main/java/mate/jdbc/Factory/FactorySingleton.java]: @EliosPride where is fix of this comment?

Comment (EliosPride) [reply src/main/java/mate/jdbc/Factory/FactorySingleton.java]: @al-pavlenko 2

**Review by al-pavlenko — APPROVED** [review]

**Pull request merged**